pkg/docker/image.go

```diff
@@ -0,0 +1,87 @@
+package docker
+
+// Create docker images with a customized docker context.
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/archive"
+
+	"go.uber.org/zap"
+)
+
+type EnsureImageOpts struct {
+	Name      string                   // reuired for EnsureImage
+	BuildCtx  string                   // required
+	BuildOpts *types.ImageBuildOptions // optional
+}
+
+func defaultBuildOptsFor(name string) *types.ImageBuildOptions {
+	return &types.ImageBuildOptions{
+		SuppressOutput: false,
+		Remove:         true,
+		ForceRemove:    true,
+		PullParent:     true,
+		Dockerfile:     "/Dockerfile",
+		Tags:           []string{name},
+	}
+}
+
+// buildImage
+// Use a Dockerfile and supporting files to build a docker image.
+// Think `docker build /path/to/build`
```

Comment:
These godocs don't follow the standard style to document a Go API; looks like loose notes. Mind revisiting?